koku/common/sql_retries.py

```diff
@@ -0,0 +1,72 @@
+import functools
+import logging
+import random
+import time
+import typing as t
+
+LOG = logging.getLogger(__name__)
+
+
+class TrinoQueryError(Exception):
+    pass
+
+
+def log_json(msg, log_ref, context):
+    return f"{msg} - {log_ref} - {context}"
+
+
+def retry(
+    original_callable: t.Callable = None,
+    *,
+    retries: int = 3,
+    retry_on: tuple[Exception, ...],
+    attribute_to_check: str = "message",
+    max_wait: int = 30,
+    context_extractor: t.Optional[t.Callable] = None,
+    log_ref: str = "Trino query",
+    custom_exception: t.Type[Exception] = None,
+    custom_log_message: t.Optional[str] = None,
+):
+    def _retry(callable: t.Callable):
+        @functools.wraps(callable)
+        def wrapper(*args, **kwargs):
+            sql_params = kwargs.get("sql_params", {})
+            if sql_params is None:
+                sql_params = {}
+            context = context_extractor(sql_params) if context_extractor else {}
```

Comment:
This is a bit leaky. Use `context` from the `kwargs` otherwise derive the context from the SQL params. If there are also no SQL params, this will be `{}`.
```suggestion
            context = kwargs.get("context", ReportDBAccessorBase.extract_context_from_sql_params(sql_params))
```